benchmarks/src/environment/snake.py

```diff
@@ -0,0 +1,64 @@
+from typing import Dict, Any
+
+import dataclasses
+import logging
+import os
+
+from pathlib import Path
+# from snakemake import snakemake
+from . import Environment, EnvironmentDescriptor
+from .utils import EnvStateManager
+
+
+@dataclasses.dataclass(frozen=True)
+class SnakeClusterInfo:
+    workdir: Path
+
+
+class SnakeEnvironmentDescriptor(EnvironmentDescriptor):
+    def create_environment(self, workdir: Path) -> Environment:
+        info = SnakeClusterInfo(workdir)
+        return SnakeEnvironment(info)
+
+    def name(self) -> str:
+        return "snake"
+
+    def parameters(self) -> Dict[str, Any]:
+        return {}
+
+    def metadata(self) -> Dict[str, Any]:
+        return {}
+
+
+class SnakeEnvironment(Environment, EnvStateManager):
+    def __init__(self, info: SnakeClusterInfo):
+        EnvStateManager.__init__(self)
+        self.info = info
+        self.snakefile = os.path.join(info.workdir, "Snakefile")
+
+    def __enter__(self):
+        self.start()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.stop()
+
+    @property
+    def workdir(self) -> Path:
+        return self.info.workdir
+
+    def start(self):
+        self.state_start()
+        logging.info("Creating Snakefile")
+        with open(self.snakefile, "w") as f:
+            pass
```

Comment:
The `start` and `stop` don't really have to do anything right now, is that correct? In that case, just call `state_start` and `state_stop` inside of them. Don't log anything nor create an empty snakefile.